### PR TITLE
Fun Phan Fixes Part 3

### PR DESCRIPTION
**Pull request by driusan**
This builds on #3535 to also fix the class of errors "PhanTypeMissingReturn"

This primarily involves changing the return comment from "none" (or similar) to "void", or fixing it to "void" on functions that don't return something when they claim to.

(The changes for this PR are entirely contained in 3a92d25d08f9178c8c74bd3f411428a55757d74f, everything else in the diff is inherited from #3535 )

### CODE REVIEW

File: modules/dataquery/php/dataquery.class.inc [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Dataquery extends \NDB_Form
     /**
      * Sets up the smarty template variables for the dataquery page.
      *
-     * @return unknown
+     * @return void
```
Comment (ZainVirani) [inline modules/dataquery/php/dataquery.class.inc]: Is this accurate? A value is returned sometimes in this function.

**Comment by driusan** [comment]
@ZainVirani Fixed them (hopefully)..

File: htdocs/survey.php [hunk 2]
```diff
@@ -38,6 +38,12 @@ class DirectDataEntryMainPage
     var $key;
 
     var $TestName;
+
```
Comment (ridz1208) [inline htdocs/survey.php]: why is this the only varialbe that got a cooment ?

File: modules/mri_violations/php/mri_protocol_check_violations.class.inc [hunk 3]
```diff
@@ -147,7 +147,7 @@ class Mri_Protocol_Check_Violations extends \NDB_Menu_Filter
      *
      * @param integer $count The offset that this page is starting at
      *
-     * @return none but as a side-effect populates $this->tpl_data['items']
+     * @return void but as a side-effect populates $this->tpl_data['items']
```
Comment (ridz1208) [inline modules/mri_violations/php/mri_protocol_check_violations.class.inc]: no parentheses ??

File: php/libraries/Database.class.inc [hunk 4]
```diff
@@ -614,13 +614,13 @@ class Database
...
      *
-     * @return     none
+     * @return     void
      * @access     public
      * @deprecated Use pselect wrappers instead
      */
     function selectRow($query, &$result)
     {
-        $this->select($query, $result);
+        $result = $this->pselect($query, array());
```
Comment (ridz1208) [inline php/libraries/Database.class.inc]: shouldn't this function be deprecated ?

is this changes related to phan fixes ?

File: htdocs/survey.php (hunk 2 above)
Comment (driusan) [reply htdocs/survey.php]: It was the only one generating an error.

Without it phan was inferring that it's a string, and generating the error:
`htdocs/survey.php:112 PhanTypeComparisonToArray string to array comparison`

File: php/libraries/Database.class.inc (hunk 4 above)
Comment (ridz1208) [reply php/libraries/Database.class.inc]: same applies below

File: php/libraries/LorisForm.class.inc [hunk 5]
```diff
@@ -2060,7 +2060,7 @@ class LorisForm
      * @param array $arr        A reference to an array to store the
      *                             extracted values into.
      *
-     * @return none, but will modify &$arr parameter passed as a side-effect
+     * @return void, but will modify &$arr parameter passed as a side-effect
```
Comment (ridz1208) [inline php/libraries/LorisForm.class.inc]: no parentheses ??

File: php/libraries/NDB_BVL_Feedback.class.inc [hunk 6]
```diff
@@ -603,7 +603,7 @@ class NDB_BVL_Feedback
     *
     * @return array
     */
-    function getThreadEntries($feedbackID)
+    static function getThreadEntries($feedbackID)
```
Comment (ridz1208) [inline php/libraries/NDB_BVL_Feedback.class.inc]: this change seems out of context

**Comment by ridz1208** [comment]
@driusan seems to be having a phpcs issue but otherwise looks good.

The only important comment is the "static" change, can you explain it ?

File: modules/mri_violations/php/mri_protocol_check_violations.class.inc (hunk 3 above)
Comment (driusan) [reply modules/mri_violations/php/mri_protocol_check_violations.class.inc]: sure, I'll add them for consistency..

File: php/libraries/Database.class.inc (hunk 4 above)
Comment (driusan) [reply php/libraries/Database.class.inc]: yeah, it should be.. but do you think 20.0 is the release to do it? It's been deprecated for a while now, so maybe it's finally time, but this change was just to fix an error from phan (I can't remember which)

File: php/libraries/NDB_BVL_Feedback.class.inc (hunk 6 above)
Comment (driusan) [reply php/libraries/NDB_BVL_Feedback.class.inc]: It's to be able to remove "PhanStaticCallToNonStatic" from the ignored list. All the calls to it are static, so the method should be static too.

File: php/libraries/Database.class.inc (hunk 4 above)
Comment (ridz1208) [reply php/libraries/Database.class.inc]: I think it should either be completely removed or ignored by phan but I dont think we should update these functions to avoid people using them in the future...

File: php/libraries/NDB_BVL_Feedback.class.inc (hunk 6 above)
Comment (ridz1208) [reply php/libraries/NDB_BVL_Feedback.class.inc]: Just confirmed it. Dismissed

File: php/libraries/Database.class.inc (hunk 4 above)
Comment (driusan) [reply php/libraries/Database.class.inc]: Oh, now I remember: it was to fix the error:

"Call to deprecated function \Database::select() defined at php/libraries/Database.class.inc:568"

so that PhanDeprecatedFunction could be removed from the ignore list.

Comment (ridz1208) [reply php/libraries/Database.class.inc]: then I guess the solution would be to remove those

Comment (driusan) [reply php/libraries/Database.class.inc]: I'll send a PR after this to remove, but I don't think it should be done as part of this PR, and I'm not convinced that we're ready with all the legacy instruments that might still be calling it. It's going to be really painful for projects to upgrade, but at least with this change we can be sure we're not calling anything that's been deprecated in the main codebase

Comment (ridz1208) [reply php/libraries/Database.class.inc]: Arguably with this change they have no more reason to stop calling these functions. I think it's been over 3 years these functions are deprecated.

We can do a very quick survey see the effect on projects
